crates/rpc-types/src/net.rs

```diff
@@ -100,3 +100,160 @@ pub struct PeerStats {
     pub banned: u32,
     pub known: u32,
 }
+
+/// Represents the connectivity state of a peer in a network, indicating the reachability and
+/// interaction status of a node with its peers.
+#[derive(Clone, Debug, PartialEq, Copy, Default)]
+#[repr(u8)]
+pub enum Connectedness {
+    /// No current connection to the peer, and no recent history of a successful connection.
+    #[default]
+    NotConnected = 0,
+
+    /// An active, open connection to the peer exists.
+    Connected = 1,
+
+    /// Connection to the peer is possible but not currently established; usually implies a past
+    /// successful connection.
+    CanConnect = 2,
+
+    /// Recent attempts to connect to the peer failed, indicating potential issues in reachability
+    /// or peer status.
+    CannotConnect = 3,
+
+    /// Connection to the peer is limited; may not have full capabilities.
+    Limited = 4,
+}
+
+impl core::fmt::Display for Connectedness {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        match self {
+            Connectedness::NotConnected => write!(f, "Not Connected"),
+            Connectedness::Connected => write!(f, "Connected"),
+            Connectedness::CanConnect => write!(f, "Can Connect"),
+            Connectedness::CannotConnect => write!(f, "Cannot Connect"),
+            Connectedness::Limited => write!(f, "Limited"),
+        }
+    }
+}
+impl From<u8> for Connectedness {
+    fn from(value: u8) -> Self {
+        match value {
+            0 => Connectedness::NotConnected,
+            1 => Connectedness::Connected,
+            2 => Connectedness::CanConnect,
+            3 => Connectedness::CannotConnect,
+            4 => Connectedness::Limited,
+            _ => Connectedness::NotConnected,
+        }
+    }
+}
+impl Into<u8> for Connectedness {
+    fn into(self) -> u8 {
+        self as u8
+    }
+}
+impl Serialize for Connectedness {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_u8(*self as u8)
+    }
+}
+
+impl<'de> Deserialize<'de> for Connectedness {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let value = u8::deserialize(deserializer)?;
+        Ok(Connectedness::from(value))
+    }
+}
```

Comment:
We should be able to remove these and place `#[derive(Serialize, Deserialize)]` above `Connectedness`.

Reply:
> We should be able to remove these and place `#[derive(Serialize, Deserialize)]` above `Connectedness`.

Yes but the tests are failling if i derive the Serialize and Deserialize

Reply:
Let me reproduce and check it out